problems/[221215] 1328. Break a Palindrome.js

```diff
@@ -5,5 +5,26 @@
  * @return {string}
  */
 const breakPalindrome = function (palindrome) {
+  if (palindrome.length === 1) return "";
 
+  const result = palindrome.replace(/[^a]/, "a");
+
+  const deduplicate = new Set(result.split(""));
+
+  if (deduplicate.size === 1) {
+    let index = 0;
+
+    for (let i = palindrome.length - 1; i >= 0; i--) {
+      if (palindrome[i] === "a") {
+        index = i;
+        break;
+      }
+    }
```

Comment:
```suggestion
  const index = palindrome
    .split("")
    .reduce((acc, cur, idx) => cur === "a" ? idx : acc, -1);
```

가장 나중에 오는 "a"의 index를 찾기 위해서 이런 방식으로도 구할 수도 있지 않을까요? `i`와 `index`라는 변수 두 개까지 만들지 않아도 동작할 수 있을 것 같아요!

Reply:
순회를 좀 더 줄여주기 위해서 위 방법을 선택했는데, 공간을 추가로 만들지 않는 것과 순회를 줄이는 것 중에 어떤 방법이 더 나을까요?

Reply:
사실 둘 중에 어떤 것이 더 연산에 효율적일 지는 잘 모르겠지만, 변수는 어쨌든 의도치않게 수정이 발생할 수 있는 것이다보니 부수효과를 일으킬 수 있는 점에서 조금 더 불안정한 방법이 아닐까 생각했습니다!